### ISSUE TITLE
Duplicate methods generated starting in 5.0.0

### ISSUE THREAD

**Issue by Jordan-Nelson** [dart-lang/ffigen#384]
### Issue
I just updated from 4.1.3 to 5.0.0 and I see several methods are being duplicated when I re-run ffigen.

### Reproduction
Below is a simplified version of my config. With this I am getting 3 copies of `CFRelease` (named `CFRelease`, `CFRelease1`, and `CFRelease2`), as well as multiple copies other methods. The same config does not generate duplicates with version 4.1.3.

Is this expected with 5.0.0?

```yaml
output: 'lib/src/ffi/macos/core_foundation.bindings.g.dart'
name: 'CoreFoundation'
description: 'Bindings to macOS & iOS CoreFoundation types'
headers:
  entry-points:
    - '/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFDictionary.h'
    - '/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFString.h'
    - '/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks/CoreFoundation.framework/Headers/CFData.h'
compiler-opts:
  - '-F/Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk/System/Library/Frameworks'
functions:
  include:
    - CFDictionaryCreate
    - CFDataCreate
    - CFStringGetCStringPtr
    - CFStringCreateWithCString
    - CFDataGetBytePtr
    - CFRelease
structs:
  include:
    - NONE
enums:
  include:
    - NONE
globals:
  include:
    - NONE
macros:
  include:
    - NONE
unions:
  include:
    - NONE
unnamed-enums:
  include:
    - NONE
comments: false
```

**Comment by liamappelbe** [comment]
Interesting. Looks like you're consuming the internal Objective C APIs, but you say you've been doing this since before 5.0.0, where experimental Objective C support was added, and you're not setting the language flag in the config. So I guess you've been parsing the C compatible structs and functions and ignoring everything else?

Anyway, presumably this has something to do with ObjC support being added. I can take a look.

You can safely ignore the duplicated methods for now though. They'll be removed by the compiler if you don't use them.

**Comment by Jordan-Nelson** [comment]
Yeah - Most of (maybe all of) CoreFoundation is C compatible.

**Comment by jerel** [comment]
I'm also seeing this starting in 5.0.0 and I'm not using ObjC. Here's my reduced configuration:

```c
/*
 * shared.h
 */
#ifndef __TYPES_INCLUDED__
#define __TYPES_INCLUDED__

uint8_t free_vec(int64_t len, const void *ptr);

#endif
```

```c
/*
 * one.h
 */
#include "../shared.h"
void some_function(int64_t value);
```

```c
/*
 * two.h
 */
#include "../shared.h"
void other_function(int64_t value);
```

And since 5.0.0 I see a bunch of numbered functions: `free_vec`, `free_vec1`, `free_vec2`, etc. Based on the number of duplicates in my project it looks like it's maybe generating the shared.h plus another copy for every time the file is included. At first glance it seems like `< 5.0.0` was respecting the `ifndef` but `=> 5.0.0` is just pulling everything in?

**Comment by dcharkes** [comment]
Multiple users are running into this. @liamappelbe could you take a look at this, you self-assigned this? Or maybe @mannprerak2 wants to take a look, since it is also happening for non-Objective.

@jerel thanks for a minimal repro! 👍

**Comment by liamappelbe** [comment]
I'll take a look.

**Comment by mannprerak2** [comment]
Upon a quick look, it seems like we accidentally used the wrong function to check if a function is seen.
```dart
bool shouldIncludeFunc(String usr, String name) {
  return _shouldIncludeDecl(
      usr, name, bindingsIndex.isSeenType, config.functionDecl.shouldInclude);
}
```
Here `bindingsIndex.isSeenType` should probably be `bindingsIndex.isSeenFunc`.

**Comment by dcharkes** [comment]
We've released v6.1.2 with the fix.
